src/v2/view-builder/views/device/DeviceChallengePollView.js

```diff
@@ -147,8 +147,12 @@ const Body = BaseFormWithPolling.extend(
 
     doCustomURI() {
       this.ulDom && this.ulDom.remove();
+      let srcUrl = this.customURI;
```

Comment:
I would create a srcUrlWithLoginHint derived property in `AppState` and would reuse it these views..

Reply:
Discussed with Pradeep, looks like we won't be able use `derived property` to retrieve username entered by user and pass to later views. But  we are going to store the value in the `settings` persistent model to be used in the later views.